index.js

```diff
@@ -127,15 +145,42 @@ Nickname ANN. 13/F witch. Best friend of Scillia. She creates all of Scillia's p
   const speedDistanceRate = 0.07;
   useFrame(({timestamp, timeDiff}) => {
     if (npcPlayer && physics.getPhysicsEnabled()) {
-      if (target) {
-        const v = localVector.setFromMatrixPosition(target.matrixWorld)
+      if (target && npcFarawayLocalPlayer()) {
+        if (localPlayerFarawayLastDest()) {
+          // console.log('localPlayerFarawayLastDest')
+
+          // Don't allowNearest here, otherwise will get inaccurate result when accurate result already exists.
+          waypointResult = pathFinder.getPath(npcPlayer.position, localPlayer.position, false);
+          setWaypointResult(waypointResult);
+        }
+
+        const isNpcReachedDest = npcReachedDest();
+        if (isNpcReachedDest) {
+          // console.log('npcReachedDest')
+          waypointResult = pathFinder.getPath(npcPlayer.position, localPlayer.position, true); // allowNearest
```

Comment:
This is a confusing API. Can we just get rid of `allowNearest`? Why should the user be determining the accuracy of the pathfind?

Reply:
Set it default `true`. https://github.com/webaverse/npc/pull/2/commits/4351daa68eb0c90d8ed84848aa6f77e39f6f8417

Reply:
Solved conflicts.